WhisperAPI/WhisperAPI/Services/NLPAPI/NlpCall.cs

```diff
@@ -29,16 +29,19 @@ public NlpCall(HttpClient httpClient, List<string> irrelevantIntents, string bas
             this.InitHttpClient();
         }
 
-        public void UpdateAndAnalyseSearchQuery(SearchQuery searchQuery)
+        public NlpAnalysis AnalyseSearchQuery(SearchQuery searchQuery)
         {
-            var nlpAnalysis = this.GetNlpAnalysis(searchQuery.Query);
+            var response = this._httpClient.PostAsync("NLP/Analyze", CreateStringContent(searchQuery.Query)).Result;
+            response.EnsureSuccessStatusCode();
+            var nlpAnalysis = JsonConvert.DeserializeObject<NlpAnalysis>(response.Content.ReadAsStringAsync().Result);
+
             if (nlpAnalysis == null)
```

Comment:
I know it is not for this PR but: we really want to return null if it can't Deserialize? We would put null in ContextItem.. I would say: throw an exception because it's not normal to receive null.